test/integration/integration_test.go

```diff
@@ -140,6 +145,40 @@ func TestProvision(t *testing.T) {
 
 	// Ensure response is equal to request cluster
 	assert.Equal(t, expectedCluster, cluster)
+
+	// The following parts tests non-existent plans not set by the administrator when provisioning
+	instanceID = uuid.New().String()
+	clusterName = brokerlib.NormalizeClusterName(instanceID)
+
+	// Setting up our Expected cluster
+	params = `{
+		"cluster": {
+			"providerSettings": {
+                "regionName": "EU_WEST_1"
+            }
+		}
+	}`
+
+	_, err = broker.Provision(ctx, instanceID, brokerapi.ProvisionDetails{
+		ServiceID:     "aosb-cluster-service-aws",
+		PlanID:        "aosb-cluster-plan-aws-m60",
+		RawParameters: []byte(params),
+	}, true)
+	assert.Error(t, atlas.ErrPlanIDNotFound, err)
+
+	_, err = broker.Provision(ctx, instanceID, brokerapi.ProvisionDetails{
+		ServiceID:     "aosb-cluster-service-gcp",
+		PlanID:        "aosb-cluster-plan-gcp-m60",
+		RawParameters: []byte(params),
+	}, true)
+	assert.Error(t, atlas.ErrPlanIDNotFound, err)
+
+	_, err = broker.Provision(ctx, instanceID, brokerapi.ProvisionDetails{
+		ServiceID:     "aosb-cluster-service-azure",
+		PlanID:        "aosb-cluster-plan-azure-m60",
+		RawParameters: []byte(params),
+	}, true)
+	assert.Error(t, atlas.ErrPlanIDNotFound, err)
```

Comment:
Can these be part of a separate integration test to keep the responsibilities clear. We also need a test for what happens if no file is provided.